allensdk/brain_observatory/ecephys/ecephys_session.py

```diff
@@ -622,7 +617,10 @@ def _build_mean_waveforms(self, mean_waveforms):
         # TODO: there is a bug either here or (more likely) in LIMS unit data ingest which causes the peak channel 
         # to be off by a few (exactly 1?) indices
         # we could easily recompute here, but better to fix it at the source
-        channel_id_lut = {(row['local_index'], row['probe_id']): cid for cid, row in self.channels.iterrows()}
+        channel_id_lut = defaultdict(
+            lambda *a, **k: -1,
```

Comment:
This is pretty gnarly and I'm not sure I understand... is this creating a dictionary where non-existent keys have a default value of `-1`?

If so, then the following would be way more readable in my opinion...
```
channel_id_lut = defaultdict(lambda: -1)
for cid, row in self.channels.iterrows():
    channel_id_lit[(row['local_index'], row['probe_id']) = cid
```